sync_labels.py

```diff
@@ -0,0 +1,207 @@
+    dst_add = src_label_names - dst_label_names
+    dst_upd = src_label_names.intersection(dst_label_names)
+    dst_del = dst_label_names - src_label_names
+
+    # Show summary of what actions will be taken
+    if len(dst_add) > 0:
+        logger.info(f"The following labels will be created in {dst_repo}:")
+        print_list(list(dst_add), " [+] ")
+
+    if len(dst_upd) > 0:
+        logger.info(f"The following labels will be updated in {dst_repo}:")
+        print_list(list(dst_upd), " [u] ")
+
+    if delete:
+        if len(dst_del) > 0:
+            logger.info(f"The following labels will be deleted in {dst_repo}:")
+            print_list(list(dst_del), " [-] ")
+
+    # Add new labels
+    add_ok = []
+    add_fail = []
+    for name in dst_add:
+        if dry_run:
+            add_ok.append(name)
+        else:
+            print_sameline("Adding label: " + name)
+            label = src.get_label(name)
+            color = label.color
+            description = label.description
+            try:
+                dst.create_label(name, color, description)
+            except GithubException:
+                logging.exception(f"Error adding label: {name}")
+                add_fail.append(name)
+            else:
+                add_ok.append(name)
+
+    # Update existing labels
+    upd_ok = []
+    upd_fail = []
+    for name in dst_upd:
+        if dry_run:
+            upd_ok.append(name)
+        else:
+            print_sameline("Updating label: " + name)
+            label = src.get_label(name)
+            dst_label = dst.get_label(name)
+            color = label.color
+            description = label.description
+            try:
+                dst_label.edit(name, color, description)
+            except GithubException:
+                logging.exception(f"Error updating label: {name}")
+                upd_fail.append(name)
+            else:
+                upd_ok.append(name)
+
+    # Delete labels
+    del_ok = []
+    del_fail = []
+    if delete:
+        for name in dst_del:
+            if dry_run:
+                del_ok.append(name)
+            else:
+                print_sameline("Deleting label: " + name)
+                label = dst.get_label(name)
+                try:
+                    label.delete()
+                except GithubException:
+                    logging.exception(f"Error deleting label: {name}")
+                    del_fail.append(name)
+                else:
+                    del_ok.append(name)
+
+    if not dry_run:
+        print_sameline("Done.", last_line=True)
+
+    if len(add_fail) > 0:
+        logger.info(f"Failed to add labels to {src_repo}:")
+        print_list(add_fail, " [!] ")
+    if len(upd_fail) > 0:
+        logger.info(f"Failed to update labels in {src_repo}:")
+        print_list(add_fail, " [!] ")
+    if len(del_fail) > 0:
+        logger.info(f"Failed to delete labels from {src_repo}:")
+        print_list(del_fail, " [!] ")
```

Comment:
The script's exit code should reflect if any errors occurred. You can do that by raising an exception or calling sys.exit(). I would prefer not doing all this tracking of errors and continuing with other labels. I'd be fine if the script simply failed on the first problem. If one label failed, I need to rerun the script anyway so continuing with other labels isn't gonna buy us anything except the complexity here. Also, if one label fails, the others are likely to fail also which means that continuing is likely pointless. So focus on correctness (program should return a non-zero exit code if failures occurred) and avoiding pointless complexity.

Reply:
removed failed lists, added exit codes.